src/lib/pci_virtio_sock.c

```diff
@@ -1020,12 +1022,13 @@ static void send_response_common(struct pci_vtsock_softc *sc,
 }
 
 static void send_response_sock(struct pci_vtsock_softc *sc,
-				 uint16_t op, uint32_t flags,
-				 const struct pci_vtsock_sock *sock)
+			       uint16_t op, uint32_t flags,
+			       struct pci_vtsock_sock *sock)
 {
 	send_response_common(sc, sock->local_addr, sock->peer_addr,
 			     op, VIRTIO_VSOCK_TYPE_STREAM, flags,
 			     sock->buf_alloc, sock->fwd_cnt);
+	sock->credit_cnt = sock->fwd_cnt;
```

Comment:
Is this the correct place for this? At this point we are just queueing the response onto the ring which goes between our local tx and rx threads, so it won't yet be visible to the driver.

Would it be more correct to defer this update to the point where the rx thread dequeues and processes it? That's the point at which the driver could be said to be about to see this update. That would mean somewhere like in ` pci_vtsock_proc_rx`, which actually it looks like you have as well?